gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/zosmf/ZosmfService.java

```diff
@@ -557,12 +557,13 @@ public JWKSet getPublicKeys() {
         final String url = getURI(getZosmfServiceId(), authConfigurationProperties.getZosmf().getJwtEndpoint());
 
         try {
-            final String json = restTemplateWithoutKeystore.getForObject(url, String.class);
-            return JWKSet.parse(json);
+            return JWKSet.load(new URL(url));
         } catch (ParseException pe) {
             log.debug("Invalid format of public keys from z/OSMF", pe);
         } catch (HttpClientErrorException.NotFound nf) {
             log.debug("Cannot get public keys from z/OSMF", nf);
+        } catch (IOException me) {
+            log.debug("Can't read JWK due to the exception " + me.getMessage(), me.getCause());
```

Comment:
should it be maybe warn? maybe consider using string interpolation for logging the message rather than concatenation